src/main/java/com/google/cloud/dns/ZoneInfo.java

```diff
@@ -49,6 +53,274 @@
+          && Objects.equals(keyLength, keySpec.keyLength)
+          && Objects.equals(keyType, keySpec.keyType);
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("algorithm", getAlgorithm())
+          .add("keyLength", getKeyLength())
+          .add("keyType", getKeyType())
+          .toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(algorithm, keyLength, keyType);
+    }
+
+    DnsKeySpec toPb() {
+      DnsKeySpec dnsKeySpecPb = new DnsKeySpec();
+      dnsKeySpecPb.setAlgorithm(algorithm);
+      dnsKeySpecPb.setKeyLength(keyLength);
+      dnsKeySpecPb.setKeyType(keyType);
+      return dnsKeySpecPb;
+    }
+
+    static KeySpec fromPb(DnsKeySpec dnsKeySpec) {
+      Builder builder = newBuilder();
+      builder.setAlgorithm(dnsKeySpec.getAlgorithm() == null ? null : dnsKeySpec.getAlgorithm());
+      builder.setKeyLength(dnsKeySpec.getKeyLength() == null ? null : dnsKeySpec.getKeyLength());
+      builder.setKeyType(dnsKeySpec.getKeyType() == null ? null : dnsKeySpec.getKeyType());
+      return builder.build();
+    }
+
+    /** Returns the DNSSEC algorithm of this key. */
+    public String getAlgorithm() {
+      return algorithm;
+    }
+
+    /** Returns the key length. */
+    public Long getKeyLength() {
+      return keyLength;
+    }
+
+    /** Returns the key type. */
+    public String getKeyType() {
+      return keyType;
+    }
+  }
+
+  /** This class represents the DNSSEC configuration. */
+  public static class DnsSecConfig {
+
+    private List<KeySpec> defaultKeySpecs;
+    private String nonExistence;
+    private String state;
+
+    public static class Builder {
+      private List<KeySpec> defaultKeySpecs;
+      private String nonExistence;
+      private String state;
+
+      private Builder() {}
+
+      private Builder(DnsSecConfig dnsSecConfig) {
+        this.defaultKeySpecs = dnsSecConfig.defaultKeySpecs;
+        this.nonExistence = dnsSecConfig.nonExistence;
+        this.state = dnsSecConfig.state;
+      }
+
+      /**
+       * Specifies parameters for generating initial DnsKeys for this ManagedZone. This can be
+       * change while state is OFF.
+       */
+      public Builder setDefaultKeySpecs(List<KeySpec> defaultKeySpecs) {
+        this.defaultKeySpecs = defaultKeySpecs;
+        return this;
+      }
+
+      /**
+       * Specifies the mechanism for authenticated denial-of-existence responses. This can be change
+       * while state is OFF. Acceptable values are 'nsec' or 'nsec3'.
+       */
+      public Builder setNonExistence(String nonExistence) {
+        this.nonExistence = nonExistence;
+        return this;
+      }
+
+      /**
+       * Specifies whether DNSSEC is enabled, and what mode it is in. Acceptable values are 'on',
+       * 'off' or 'transfer'.
+       */
+      public Builder setState(String state) {
+        this.state = state;
+        return this;
+      }
```

Comment:
The same, what if the state is not acceptable? Would it make sense to add check and throw IllegalaArgumentException?

Reply:
> The same, what if the state is not acceptable?

Throw DnsException with bad request -> status -> INVALID_ARGUMENT

> Would it make sense to add check and throw IllegalaArgumentException?

yes, it's much better to prevent before the actual Api call